framework/src/play/data/validation/ValidationPlugin.java

```diff
@@ -108,14 +113,19 @@ static Validation restore(Request request) {
             Validation validation = new Validation();
             Http.Cookie cookie = request.cookies.get(Scope.COOKIE_PREFIX + "_ERRORS");
             if (cookie != null) {
-                String errorsData = URLDecoder.decode(cookie.value, "utf-8");
-                Matcher matcher = errorsParser.matcher(errorsData);
-                while (matcher.find()) {
-                    String[] g2 = matcher.group(2).split("\u0001", -1);
-                    String message = g2[0];
-                    String[] args = new String[g2.length - 1];
-                    System.arraycopy(g2, 1, args, 0, args.length);
-                    validation.errors.add(new Error(matcher.group(1), message, args));
+                String decryptErrors =   errorsCookieCrypter.decrypt(cookie.value);
+                if (decryptErrors != null) {
+                    String errorsData = URLDecoder.decode(decryptErrors, "utf-8");
```

Comment:
I guess the order is wrong: you first need to decode and then to decrypt.

Reply:
refactored